GVRf/Framework/framework/src/main/jni/engine/picker/picker_jni.cpp

```diff
@@ -104,24 +109,43 @@ Java_org_gearvrf_NativePicker_pickObjects(JNIEnv * env,
     return pickList;
 }
 
-JNIEXPORT jfloat JNICALL
+JNIEXPORT jobject JNICALL
 Java_org_gearvrf_NativePicker_pickSceneObject(JNIEnv * env,
-        jobject obj, jlong jscene_object, jlong jcamera_rig) {
+                                              jobject obj, jlong jscene_object,
+                                              jfloat ox, jfloat oy, jfloat oz,
+                                              jfloat dx, jfloat dy, jfloat dz) {
     SceneObject* scene_object =
             reinterpret_cast<SceneObject*>(jscene_object);
-    CameraRig* camera_rig = reinterpret_cast<CameraRig*>(jcamera_rig);
-    return Picker::pickSceneObject(scene_object, camera_rig);
+
+    ColliderData data;
+    Picker::pickSceneObject(scene_object, ox, oy, oz, dx, dy, dz, data);
+
+    jclass pickerClass = env->FindClass("org/gearvrf/GVRPicker");
+    jclass hitClass = env->FindClass("org/gearvrf/GVRPicker$GVRPickedObject");
```

Comment:
I know there is already code doing it but FindClass and GetMethod are not free. Please change the code to cache the results of these lookups.